### PR TITLE
Feature/table bulk actions deselect lines on input search clear

### PR DESCRIPTION
**Pull request by ericreis**
#### What is the purpose of this pull request?
Deselect all lines on input search clear.

#### What problem is this solving?
Prevents one possible wrong behavior mentioned in #873.

#### How should this be manually tested?
Last example of http://10.1.14.209:6060/#/Components/Display/Table

#### Screenshots or example usage
N/A

### CODE REVIEW

File: react/components/Table/index.js [hunk 1]
```diff
@@ -250,12 +255,31 @@ class Table extends PureComponent {
...
+                ...toolbar.inputSearch,
+                onClear: hasBulkActions
+                  ? this.handleInputSearchClearWithBulkAction
+                  : toolbar.inputSearch.onClear,
+              }
+            : null,
+        }
+      : null
+
+    console.log(extendedToolbar)
```
Comment (ohmyguigs) [inline react/components/Table/index.js]: spoted
```suggestion
```

File: react/components/Table/index.js [hunk 2]
```diff
@@ -250,12 +255,31 @@ class Table extends PureComponent {
...
+
```
Comment (ohmyguigs) [inline react/components/Table/index.js]:
```suggestion
```

File: react/components/Table/index.js [hunk 3]
```diff
@@ -139,6 +139,11 @@ class Table extends PureComponent {
     )
   }
 
+  handleInputSearchClearWithBulkAction = event => {
+    this.handleDeselectAllLines()
+    this.props.toolbar.inputSearch.onClear(event)
```
Comment (kevinch) [inline react/components/Table/index.js]: No need to check if the function exists?

Comment (ericreis) [reply react/components/Table/index.js]: We are already checking if the inputSearch exists in line 262 (265 to be more specific). The handleInputSearchClearWithBulkActions function in only being used when there are bulk actions and the input search.

Do you think it’s necessary to check if the onCleat function is defined there too?

Comment (kevinch) [reply react/components/Table/index.js]: If there's any chance of the function `onClear()` to be called without being defined then yes. Otherwise no.

**Pull request merged**